frontend/src/js/components/login/Signup.tsx

```diff
@@ -67,6 +67,10 @@ const useStyles = makeStyles()(theme => ({
   },
   logo: { marginLeft: '5vw', marginTop: 45, maxHeight: 50 }
 }));
+const getCurrentLocation = (location: Location) => {
+  const currentLocationKey = Object.keys(locations).find(key => [`staging.${locations[key].location}`, locations[key].location].includes(location.hostname));
```

Comment:
```suggestion
  const { key: currentLocationKey } = Object.values(locations).find(value => [`staging.${value.location}`, value.location].includes(location.hostname));
```

since you use the key to access the `locations` value or

```suggestion
  const [currentLocationKey] = Object.entries(locations).find(([key, value]) => [`staging.${value.location}`, value.location].includes(location.hostname));
```

if you want to rely on the key instead of the key attribute

Reply:
Both proposed solutions are type unsafe (would cause TypeError if `find` returns `undefined`), but your comment helped me find issue with my solution, thanks

Reply:
Then use a variant that is both type safe + a proper use of the object properties instead:
```suggestion
  const currentLocationKey = 
Object.entries(locations).reduce(
      (accu, [key, value]) => ([`staging.${value.location}`, value.location].includes(window.location.hostname) ? key : accu),
      locations.us.key
    )
```
If you iterate on keys, you shouldn't rely on the values.